examples/shopify/lib/shopify/api.ts

```diff
@@ -80,7 +80,7 @@ export async function getCollections(): Promise<Collection[]> {
   return result.data.collections.edges.map(edge => edge.node)
 }
 
-export async function getProduct(handle?: string): Promise<ProductFragment> {
+export async function getProduct(handle?: string): Promise<ProductFragment | null> {
```

Comment:
This was mis typed before. Tested it out and the api returns nulls as the product if you don't pass a valid handle.